checker/src/abstract_value.rs

```diff
@@ -2691,6 +2751,62 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         self.get_cached_interval()
     }
 
+    #[logfn_inputs(TRACE)]
+    fn get_cached_octagon(&self) -> Rc<OctagonDomain> {
+        {
+            let mut cached_octagon = self.octagon.borrow_mut();
+            let octagon_opt = cached_octagon.as_ref();
+            if let Some(octagon) = octagon_opt {
+                return octagon.clone();
+            }
+            let octagon = self.get_as_octagon();
+            *cached_octagon = Some(Rc::new(octagon));
+        }
+        self.get_cached_octagon()
+    }
+
+    /// Constructs an element of the Octagon domain for simple expressions.
+    #[logfn_inputs(TRACE)]
+    fn get_as_octagon(&self) -> OctagonDomain {
+        if self.expression_size > k_limits::MAX_EXPRESSION_SIZE / 10 {
+            return octagon_domain::BOTTOM;
+        }
+        match &self.expression {
+            Expression::Top => octagon_domain::BOTTOM,
+            Expression::CompileTimeConstant(ConstantDomain::I128(val)) => (*val).into(),
+            Expression::CompileTimeConstant(ConstantDomain::U128(val)) => (*val).into(),
+            Expression::Add { left, right } => {
+                left.get_as_octagon().add(&right.get_as_octagon())
+            },
+            Expression::ConditionalExpression {
+                consequent,
+                alternate,
+                ..
+            } => consequent
+                .get_as_octagon()
+                .widen(&alternate.get_as_octagon()),
+            Expression::Join { left, right, .. } => {
+                left.get_as_octagon().widen(&right.get_as_octagon())
+            },
+            Expression::Mul { left, right } => left.get_as_octagon().mul(&right.get_as_octagon()),
+            Expression::Neg { operand } => {
+                let octa = operand.get_as_octagon().neg();
+                octa
+            }
+            Expression::Sub { left, right } => left.get_as_octagon().sub(&right.get_as_octagon()),
+            Expression::Switch { cases, default, .. } => cases
+                .iter()
+                .fold(default.get_as_octagon(), |acc, (_, result)| {
+                    acc.widen(&result.get_as_octagon())
+                }),
+            _ => {
+                octagon_domain::BOTTOM
```

Comment:
This should be BOTTOM only when self is BOTTOM. How are you going to model variables? More specifically, how are you going to deal with expressions involving more variables than the domain can support?

Reply:
That's a good  question, I don't clearly see how to model it yet. My initial plan was to make the simplest version of the OD and then gradually extend it. Does it sound ok to return to the question once the OD will work well for the simple scenarios?

Reply:
Without variables the constant domain will take precedence and nothing interesting will happen in the octal domain. You could, however, get started (locally) by simply assuming there are at most two variables and cut your teeth on test cases that have at most two variables. Eventually, you'll probably have to track which variables are in the domain and go to top when a transfer function attempts to include more than two.

Reply:
I see. It perhaps better to model variables right away. I will think about how to address it.